sdk/communication/azure-communication-identity/src/main/java/com/azure/communication/identity/CommunicationIdentityAsyncClient.java

```diff
@@ -113,20 +115,57 @@ public Mono<Response<CommunicationUserIdentifier>> createUserWithResponse() {
      * Creates a new CommunicationUserIdentifier with token.
      *
      * @param scopes The list of scopes for the token.
+     * @param tokenExpiresAfter Custom validity period of the Communication Identity access token within [1,24]
+     * hours range. If not provided, the default value of 24 hours will be used.
      * @return The created communication user and token.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<CommunicationUserIdentifierAndToken>
-        createUserAndToken(Iterable<CommunicationTokenScope> scopes) {
+        createUserAndToken(Iterable<CommunicationTokenScope> scopes, Duration tokenExpiresAfter) {
         try {
             Objects.requireNonNull(scopes);
             final List<CommunicationTokenScope> scopesInput = StreamSupport.stream(scopes.spliterator(), false).collect(Collectors.toList());
-            return client.createAsync(new CommunicationIdentityCreateRequest().setCreateTokenWithScopes(scopesInput))
+
+            CommunicationIdentityCreateRequest communicationIdentityCreateRequest = new CommunicationIdentityCreateRequest();
+            communicationIdentityCreateRequest.setCreateTokenWithScopes(scopesInput);
+
+            int expiresInMinutes;
+
+            if (tokenExpiresAfter != null) {
+                try {
+                    expiresInMinutes = Math.toIntExact(tokenExpiresAfter.toMinutes());
+                }
+                catch (ArithmeticException ex){
+                    IllegalArgumentException expiresAfterOverflowEx = new IllegalArgumentException(OVERFLOW_MESSAGE, ex);
+                    return monoError(logger, expiresAfterOverflowEx);
+                }
+                communicationIdentityCreateRequest.setExpiresInMinutes(expiresInMinutes);
+            }
+
+            return client.createAsync(communicationIdentityCreateRequest)
                 .onErrorMap(CommunicationErrorResponseException.class, e -> translateException(e))
                 .flatMap(
                     (CommunicationIdentityAccessTokenResult result) -> {
                         return Mono.just(userWithAccessTokenResultConverter(result));
                     });
+        }
+        catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Creates a new CommunicationUserIdentifier with token.
+     *
+     * @param scopes The list of scopes for the token.
+     * @return The created communication user and token.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<CommunicationUserIdentifierAndToken>
+        createUserAndToken(Iterable<CommunicationTokenScope> scopes) {
+        try {
+            Objects.requireNonNull(scopes);
```

Comment:
double check, same for `createUserAndTokenWithResponse(Iterable<CommunicationTokenScope> scopes)`,  `getTokenWithResponse(CommunicationUserIdentifiercommunicationUser,
 Iterable<CommunicationTokenScope> scopes)`,
`getToken(CommunicationUserIdentifiercommunicationUser,
 Iterable<CommunicationTokenScope> scopes)`,
`createUserAndTokenWithResponse(Iterable<CommunicationTokenScope> scopes)`

Reply:
Removed.